### PR TITLE
Feature/Waffle Conditional Routing to Ember App[PLAT-692]

### PR DESCRIPTION
**Pull request by pattisdr**
## Purpose
Figure out how to conditionally route to Ember app if the corresponding flag is active for the given user.

❗️ Note: I recommend just using flags for everything, rather than a mix of switches and flags, as the ticket described.  I believe this is actually simpler.  Flags have an `everyone` boolean field. This setting overrides all other potential toggles on the flag, making it behave identically to a switch.   For example, I set 'everyone' to True, and it is active for everyone, all the time.  This way we just have to deal with the `Flag` model.

❗️Note: specific ember flags that are being added to the db should be verified by Embosf to make sure they match with the views they are planning on implementing in Ember. See google doc on ticket.

## Changes
- Upgrades django-waffle to 0.13 which has some bugfixes
- Used naming convention `ember_<page_name>_page` for waffle flags
- Created a decorator for the flask views called `ember_flag_is_active`.  Pass in the flag name.  If ember flag is active, calls the `use_ember_app` view.  Otherwise, calls the specified view.
Sample usage:
```
@ember_flag_is_active('ember_support_page')
def support():
    return {}

```
- Created a data migration to add flags for the pages specified in [Product Page:EmbOSF](https://openscience.atlassian.net/wiki/spaces/PRODUC/pages/56451563/Product+Page+EmbOSF) 
- Added decorator to views for pages on list to be emberized.

## QA Notes

Probably no testing needed, this is in support of Embosf.

## Ticket

https://openscience.atlassian.net/browse/PLAT-692

### CODE REVIEW

File: osf/migrations/0081_add_ember_waffle_flags.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    This migration adds some waffle flags for pages that are being emberized.
+    Waffle flags are used for feature flipping, for example, showing an
+    emberized page to one set of users, and the existing osf page for another set.
+
+    By default, flags are given an everyone=False value, which overrides all other settings,
+    making the flag False for everyone.  Flag settings can be changed in the Django admin app.
+    """
+    for page in EMBER_WAFFLE_PAGES:
+        try:
+            with transaction.atomic():
```
Comment (sloria) [inline osf/migrations/0081_add_ember_waffle_flags.py]: Minor: This shouldn't be necessary; each migration is run in it's own transaction by default, so this would actually create a savepoint rather than start a new transaction.

File: osf/migrations/0081_add_ember_waffle_flags.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                Flag.objects.create(name=format_ember_waffle_flag_name(page), everyone=False)
+        except IntegrityError as e:
+            # just in case someone has already added a flag with this name to the db
```
Comment (sloria) [inline osf/migrations/0081_add_ember_waffle_flags.py]: Drive-by: Can you use `get_or_create`?

File: osf/migrations/0081_add_ember_waffle_flags.py (hunk 1 above)
Comment (pattisdr) [reply osf/migrations/0081_add_ember_waffle_flags.py]: I added this because if there was already a flag in that db that existed it would throw a fatal IntegrityError.   But using get_or_create like you mentioned above will fix this problem - thanks :)

File: framework/auth/decorators.py [hunk 3]
```diff
@@ -80,6 +83,21 @@ def wrapped(*args, **kwargs):
 
     return wrapped
 
+def ember_flag_is_active(flag_name):
```
Comment (sloria) [inline framework/auth/decorators.py]: I don't think this belongs in `framework.auth`. How about moving it to `website.ember_osf_web.views` or `website.ember_osf_web.decorators`?

Comment (pattisdr) [reply framework/auth/decorators.py]: good point, moved.
